include/deal.II/base/vectorization.h

```diff
@@ -169,6 +169,29 @@ class VectorizedArrayIterator
     return *this;
   }
 
+  /**
+   * This operator advances the iterator by @p offet lanes and returns a
+   * reference to <tt>*this</tt>.
+   */
+  VectorizedArrayIterator<T> &
+  operator+=(unsigned int offset)
+  {
+    lane += offset;
```

Comment:
What do you do if `lane` is larger than `T::n_array_elements`?

Reply:
@masterleinad I don't see a problem that `lane` is outside of the range `[0;T::size()[`. In contrast, we need at least `static_cast<unsignd int>(-1)` () and `T::size()` (in the case of incrementing) as `end()`. The problem is only when we want to dereferene the iterator. This is why I have added an assert to `operator*()`. Do you agree that this enough?

Reply:
Yes, I meant to say that we should make sure that the element referenced stays valid. I think checking in the accessor is fine.

Reply:
I'm usually of the opinion that we should do the checking at the first possible place because it makes debugging so much simpler. So I would advocate putting an assertion here as well.